hy/lex/hy_reader.py

```diff
@@ -0,0 +1,456 @@
+            raise LexException.from_reader(
+                "Cannot access attribute on anything other"
+                " than a name (in order to get attributes of expressions,"
+                " use `(. <expression> <attr>)` or `(.<attr> <expression>)`)",
+                self,
+            )
+        return Keyword(ident, from_parser=True)
+
+    @reader_for('"')
+    def prefixed_string(self, _, prefix=""):
+        escaping = False
+
+        def quote_closing(c):
+            nonlocal escaping
+            if c == "\\":
+                escaping = not escaping
+                return 0
+            if c == '"' and not escaping:
+                return 1
+            escaping = False
+            return 0
+
+        return self.read_string_until(quote_closing, prefix, "f" in prefix.lower())
+
+    ###
+    # Special annotations
+    ###
+
+    @reader_for("'", ("quote",))
+    @reader_for("`", ("quasiquote",))
+    def tag_as(root):
+        def _tag_as(self, _):
+            nc = self.peekc()
+            if not nc or nc.isspace() or self.reader_table.get(nc) == self.INVALID:
+                raise LexException.from_reader(
+                    "Could not identify the next token.", self
+                )
+            node = self.parse_one_node()
+            return mkexpr(root, node)
+
+        return _tag_as
+
+    @reader_for("~")
+    def unquote(self, key):
+        nc = self.peekc()
+        if not nc or nc.isspace() or self.reader_table.get(nc) == self.INVALID:
+            return sym(key)
+        if self.peek_and_getc("@"):
+            root = "unquote-splice"
+        else:
+            root = "unquote"
+        node = self.parse_one_node()
+        return mkexpr(root, node)
+
+    @reader_for("^")
+    def annotate_or_xor(self, _):
+        suffix = self.read_ident(just_peeking=True)
+        if suffix == "":
+            return sym("^")
+        if suffix == "=":
+            self.getc()
+            return sym("^=")
+        node = self.parse_one_node()
+        return mkexpr("annotate", node)
+
+    ###
+    # Sequences
+    ###
+
+    @reader_for("(", (Expression, ")"))
+    @reader_for("[", (List, "]"))
+    @reader_for("{", (Dict, "}"))
+    @reader_for("#{", (Set, "}"))
+    def sequence(seq_type, closer):
+        def _sequence(self, _):
+            return seq_type(self.parse_nodes_until(closer))
+
+        return _sequence
+
+    ###
+    # Reader tag-macros
+    ###
+
+    @reader_for("#")
+    def tag_dispatch(self, key):
+        """General handler for reader macros (and tag macros).
+
+        Reads a full identifier after the `#` and calls the corresponding handler
+        (this allows, e.g., `#reads-multiple-forms foo bar baz`).
+
+        Failing that, reads a single character after the `#` and immediately
+        calls the corresponding handler (this allows, e.g., `#*args` to parse
+        as `#*` followed by `args`).
+
+        Failing that, parses the text as an old-style tag macro (e.g., `#ident
```

Comment:
Minor nit, this part of the docstring is no longer true

Reply:
fixed